File: onecard-gameserver/src/main/kotlin/model/GameState.kt
package model

import kotlinx.serialization.Serializable


enum class Direction {
    CLOCKWISE, COUNTERCLOCKWISE
}

enum class GameStatus {
    WAITING, PLAYING
}

@Serializable
data class GameState (
    val players: List<String>,
    val currentPlayerIndex: Int,
    val deck: List<PokerCard>,
    val discardPile: List<PokerCard>,
    val direction: Direction,
    val damage : Int,
    val gameStatus: GameStatus,
    val setting : Settings,
)

@Serializable
data class Settings (
    val numberOfPlayers: Int,
    val includeJokers: Boolean,
    val initHandSize: Int,
    val maxHandSize: Int,
)

File: onecard-client/types/gameState.ts
import { Player } from './gamePlayer';
import { PokerCardPropsWithId } from './pokerCard';

// 게임 상태 및 설정 관련 정의
export type Direction = 'clockwise' | 'counterclockwise';
export type Mode = 'single' | 'multi';
export type GameStatus = 'waiting' | 'playing' | 'finished';

export interface GameSettings {
	mode: Mode;
	numberOfPlayers: number;
	includeJokers: boolean;
	initHandSize: number;
	maxHandSize: number;
}

export interface GameState {
	players: Player[];
	currentPlayerIndex: number;
	deck: PokerCardPropsWithId[];
	discardPile: PokerCardPropsWithId[];
	direction: Direction;
	damage: number;
	gameStatus: GameStatus;
	settings: GameSettings;
	winner?: Player;
}
